ui/src/core/app_impl.ts

```diff
@@ -197,6 +199,34 @@ export class AppImpl implements App {
     };
   }
 
+  get zenModeEnabled(): boolean {
+    return this._zenModeEnabled;
+  }
+
+  get topbarVisibleInZenMode(): boolean {
+    return this._topbarVisibleInZenMode;
+  }
+
+  showTopbarInZenMode(): void {
+    this._topbarVisibleInZenMode = true;
+    raf.scheduleFullRedraw();
```

Comment:
I'd probably omit these `raf.scheduleFullRedraw();` calls entirely - here and below